Automattic/woocommerce-payments
# Ensure shipping is re-added after payment request is aborted

Fixes #8097 

#### Changes proposed in this Pull Request

The solution in this PR currently is to re-initialize the payment request if a previous payment request has been aborted and the product quantity has changed. This is similar to how the same problem is handled with switching variations after an aborted payment request.

The reason that shipping isn't re-added when adjusting quantity after an aborted payment request comes down to the payment request update that is performed using data from `get_selected_product_data()`. Here the shipping item isn't calculated and `pending` is set to `true`, which tells the payment sheet to not display this item. 

https://github.com/Automattic/woocommerce-payments/blob/c4f63dba07b17a92a4c309ba1c21e0b3e1d6996b/includes/class-wc-payments-payment-request-button-handler.php#L1023-L1036











#### Testing instructions


1. Enabled Payment Requests
2. Add a shipping method that calculates dynamically, such as a flat rate with a value of `[fee percent="10"]`.
3. Navigate to a product and attempt to checkout via Google Pay, confirming that shipping appears in the order details inside the payment request sheet
4. Close the sheet
5. Increase the product quantity
6. Click the Google Pay button once more and check that shipping still appears and the cost is correct





-------------------


**Post merge**
